packages/compiler_dart/lib/src/builtins.dart

```diff
@@ -122,9 +123,146 @@ class DartBuiltinCompiler extends BuiltinCompiler<dart.Spec> {
+        .expand((values) sync* {
+      final id = values.first;
+      final function = values.second;
+
+      if (function.isStatic) {
+        throw CompilerError.unsupportedFeature(
+          'Static functions in impls are not yet supported.',
+          location: ErrorLocation(
+            id.resourceId,
+            getPropertyDeclarationAst(context, id)
+                .modifiers
+                .firstWhere((w) => w is StaticModifierToken)
+                .span,
+          ),
+        );
+      }
+
+      yield dart.Method((b) => b
+        ..annotations.add(dart.refer('override', dartCoreUrl))
+        ..returns = compileType(context, function.returnType)
+        ..name = function.name
+        ..types.addAll(function.typeParameters
+            .map((it) => compileTypeParameter(context, it)))
+        ..requiredParameters
+            .addAll(compileParameters(context, function.valueParameters))
+        ..body = compileBody(context, id).value);
+    });
+
+    final t = dart.refer('T');
+    final arrayT = dart.refer('Array<T>');
+    final listT = dart.TypeReference((b) => b
+      ..symbol = 'List'
+      ..url = dartCoreUrl
+      ..types.add(t));
+    return [
+      dart.Class((b) => b
+        ..annotations.add(dart.refer('sealed', packageMetaUrl))
+        ..name = 'Array'
+        ..types.add(t)
+        ..fields.add(dart.Field((b) => b
+          ..name = 'value'
+          ..type = listT))
+        ..mixins.addAll(traits.map((it) {
+          final type = compileType(context, it);
+          return dart.TypeReference((b) => b
+            ..symbol = '${type.symbol}\$Default'
+            ..types.addAll(it.arguments.map((it) => compileType(context, it)))
+            ..url = type.url);
+        }))
+        ..implements.addAll(implements)
+        ..constructors.add(dart.Constructor((b) => b
+          ..requiredParameters
+              .add(dart.Parameter((b) => b..name = 'this.value'))))
+        ..methods.addAll([
+          dart.Method((b) => b
+            ..name = 'generate'
+            ..static = true
+            ..types.add(t)
+            ..returns = arrayT
+            ..requiredParameters.addAll([
+              dart.Parameter((b) => b
+                ..name = 'length'
+                ..type = compileType(context, CandyType.int)),
+              dart.Parameter((b) => b
+                ..name = 'generator'
+                ..type = dart.FunctionType((b) => b
+                  ..requiredParameters.add(compileType(context, CandyType.int))
+                  ..returnType = t)),
+            ])
+            ..body = arrayT.call([
+              listT.property('generate').call([
+                dart.refer('length.value'),
+                // The Dart code generator doesn't support lambdas, so we do an ugly workaround.
+                dart.refer('(index) => generator').call([
+                  compileType(context, CandyType.int)
+                      .call([dart.refer('index')])
+                ]),
```

Comment:
It actually does: `dart.Method(…).closure` gives you a `dart.Expression` (which you already use at the end of this file). But we can leave it here and save the time.